hashintel/hash
# Add new `Table` block (MVP supporting local data only)

## 🌟 What is the purpose of this PR?

This is the first PR for the new Table block. It will be follow up PRs implementing new features like query building, linking data, filtering, sorting etc. This PR just introduces the most basic functionality and the usage of `glide-data-grid`.

Here is the list of the supported functionalities for now

- local columns
- local rows
- adding colums / rows
- editing cells
- updating table title
- readonly mode

## 🔗 Related links

- [Asana task](https://app.asana.com/0/1200211978612931/1203179653350866/f) _(internal)_

## 🐾 Next steps

Implement rest of the features for Table block

## ❓ How to test this?

1.  Run `table-v2` block on your local
2. Confirm that it works as expected

## 📹 Demo

⚠️ Showing copy-paste using keyboard shortcuts at the beginning of the demo


https://user-images.githubusercontent.com/39553853/223970783-8c182290-6d1a-4027-9e56-214b01e6bb13.mov



---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1203922830586217
